src/public/javascripts/ssoAuthES6.js

```diff
@@ -0,0 +1,117 @@
+        // and errors returned from the call of `getAccessToken` above.
+        if (exception.code) { 
+            handleClientSideErrors(exception);
+        }
+        else {
+            showMessage("EXCEPTION: " + JSON.stringify(exception));
+        } 
+    }
+}
+
+async function getGraphToken(bootstrapToken) {
+    let response = await $.ajax({type: "GET", 
+		url: "/auth",
+        headers: {"Authorization": "Bearer " + bootstrapToken }, 
+        cache: false
+    });
+    return response;
+}
+
+function handleClientSideErrors(error) {
+    switch (error.code) {
+
+        case 13001:
+            // No one is signed into Office. If the add-in cannot be effectively used when no one 
+            // is logged into Office, then the first call of getAccessToken should pass the 
+            // `allowSignInPrompt: true` option.
+            showMessage("No one is signed into Office. But you can use many of the add-ins functions anyway. If you want to log in, press the Get OneDrive File Names button again.");  
+            break;
+        case 13002:
+            // The user aborted the consent prompt. If the add-in cannot be effectively used when consent
+            // has not been granted, then the first call of getAccessToken should pass the `allowConsentPrompt: true` option.
+            showMessage("You can use many of the add-ins functions even though you have not granted consent. If you want to grant consent, press the Get OneDrive File Names button again."); 
+            break;
+        case 13006:
+            // Only seen in Office on the Web.
+            showMessage("Office on the Web is experiencing a problem. Please sign out of Office, close the browser, and then start again."); 
+            break;
+        case 13008:
+            // Only seen in Office on the Web.
+            showMessage("Office is still working on the last operation. When it completes, try this operation again."); 
+            break;
+        case 13010:
+            // Only seen in Office on the Web.
+            showMessage("Follow the instructions to change your browser's zone configuration.");
+            break;
+        default:
+            // For all other errors, including 13000, 13003, 13005, 13007, 13012, and 50001, fall back
+            // to non-SSO sign-in.
+            dialogFallback();
+            break;
+    }
+}
+
+function handleAADErrors(exchangeResponse) {
+    // On rare occasions the bootstrap token is unexpired when Office validates it,
+    // but expires by the time it is sent to AAD for exchange. AAD will respond
+    // with "The provided value for the 'assertion' is not valid. The assertion has expired."
+    // Retry the call of getAccessToken (no more than once). This time Office will return a 
+    // new unexpired bootstrap token. 
+    if ((exchangeResponse.error_description === "The provided value for the 'assertion' is not valid. The assertion has expired.")
```

Comment:
Do not compare the description as this can be changed at any time. Is there a specific error code such as AADSTS500133 we can look for?

Reply:
This is weird. Today I am getting that number in the error_description. It wasn't there a few days ago when I wrote this code. And the rest of the description is now worded differently too: 
"AADSTS500133: Assertion is not within its valid time range.\r\nTrace ID: 9ef9d92a-72c1-4c58-97c9-1a02212c2200\r\nCorrelation ID: dc1683b8-83e6-46e3-9974-da98c8fa34ee\r\nTimestamp: 2019-10-31 00:56:24Z"

Anyway, I've changed it to use the number